### PR TITLE
Added methods which get cluster list, topic identities, and partition identities

### PR DESCRIPTION
**Pull request by khhirani**
Added the following getter methods to kafka_manager_api.py: get_cluster_list and get_topic_identities. get_cluster_list allows user to filter for active or pending clusters. Added tests to kafka_manager_api_test for these functions.

### CODE REVIEW

File: test/kafka_manager_api_test.py [hunk 1]
```diff
@@ -88,12 +105,55 @@ def test_get_kafka_api_no_inputs(self, mock_kafka_manager, mock_cli_args, mock_p
...
+    @patch('krux_kafka_manager.kafka_manager_api.get_parser')
+    @patch('krux_kafka_manager.kafka_manager_api.add_kafka_manager_api_cli_arguments')
+    @patch('krux_kafka_manager.kafka_manager_api.KafkaManagerAPI')
+    def test_get_kafka_api_no_inputs(self, mock_kafka_manager, mock_cli_args, mock_parser, mock_logger, mock_stats):
+        """
+        Kafka Manager API Test: Checks if get_kafka_manager_api initalizes KafkaManagerAPI object
+        with no user inputs provided (except mandatory hostname argument)
+        """
+        mock_parser.return_value.parse_args.return_value = MagicMock(hostname=KafkaManagerTest._HOSTNAME)
+        mock_kafka_manager.return_value = MagicMock()
```
Comment (hanpeter) [inline test/kafka_manager_api_test.py]: Nit: MagicMock always return another MagicMock. Technically, this is unnecesary.

File: test/kafka_manager_api_test.py [hunk 2]
```diff
@@ -88,12 +105,55 @@ def test_get_kafka_api_no_inputs(self, mock_kafka_manager, mock_cli_args, mock_p
             stats=mock_stats(prefix=NAME),
             )
 
+    @patch('krux_kafka_manager.kafka_manager_api.get_stats')
+    @patch('krux_kafka_manager.kafka_manager_api.get_logger')
+    @patch('krux_kafka_manager.kafka_manager_api.get_parser')
+    @patch('krux_kafka_manager.kafka_manager_api.add_kafka_manager_api_cli_arguments')
+    @patch('krux_kafka_manager.kafka_manager_api.KafkaManagerAPI')
+    def test_get_kafka_api_no_inputs(self, mock_kafka_manager, mock_cli_args, mock_parser, mock_logger, mock_stats):
```
Comment (hanpeter) [inline test/kafka_manager_api_test.py]: You are mocking logger and stats but not using it here. Why?

File: test/kafka_manager_api_test.py [hunk 3]
```diff
@@ -27,17 +27,33 @@
 class KafkaManagerTest(unittest.TestCase):
 
     _HOSTNAME = 'test_hostname'
+    CLUSTER = 'cluster'
+    TOPIC = 'topic_name'
+    STATUS = 'active'
+    CLUSTER_LIST_REQUEST = {'clusters':[{'name': 'cluster1', 'status': 'active'}]}
```
Comment (hanpeter) [inline test/kafka_manager_api_test.py]: Link this with `CLUSTER_LIST_RV` like this: `CLUSTER_LIST_REQUEST = {'clusters': CLUSTER_LIST_RV}`. This way, when I change `CLUSTER_LIST_RV`, `CLUSTER_LIST_REQUEST` will changed also.

File: test/kafka_manager_api_test.py [hunk 4]
```diff
@@ -27,17 +27,33 @@
...
+    CLUSTER_LIST_STATUS = {
```
Comment (hanpeter) [inline test/kafka_manager_api_test.py]: Link this with `CLUSTER_LIST_RV` like this: `CLUSTER_LIST_REQUEST = {'clusters': CLUSTER_LIST_RV}`. This way, when I change `CLUSTER_LIST_RV`, `CLUSTER_LIST_REQUEST` will changed also.

File: test/kafka_manager_api_test.py [hunk 5]
```diff
@@ -27,17 +27,33 @@
...
+            'clusters': { 
+                'active': [{'name': 'cluster1', 'status': 'active'}],
+                'pending': [{'name': 'cluster2', 'status': 'pending'}]
+            }
+        }
+    CLUSTER_LIST_RV =  [{'name': 'cluster1', 'status': 'active'}]
+    TOPIC_IDENTITIES_REQUEST = {'topicIdentities': [{'topic':'topic_name', 'partitionsIdentity':
```
Comment (hanpeter) [inline test/kafka_manager_api_test.py]: Refer to above.

File: test/kafka_manager_api_test.py (hunk 4 above)
Comment (hanpeter) [inline test/kafka_manager_api_test.py]: Also use `STATUS`.

File: test/kafka_manager_api_test.py (hunk 2 above)
Comment (khhirani) [inline test/kafka_manager_api_test.py]: I use them on L123-124 to make sure they were called with the correct name/prefix in the constructor.

Comment (hanpeter) [inline test/kafka_manager_api_test.py]: Ah, sorry. My mistake.

**Comment by hanpeter** [comment]
LGTM

**Comment by dcrampton** [comment]
Did the tests that you wrote behave as expected? If so, +1.

**Pull request merged**